config.go

```diff
@@ -85,6 +88,11 @@ type neutrinoConfig struct {
 	BanThreshold uint32        `long:"banthreshold" description:"Maximum allowed ban score before disconnecting and banning misbehaving peers."`
 }
 
+type coreConfig struct {
```

Comment:
I've held off on making any changes on this until we clear up the naming and grouping.

I've named it "core mode" so that it's obvious that it's using Bitcoin Core rather than btcd. My intention is to reuse this for other cryptocurrencies in the future: since we have ltcd support, we can have Litecoin Core node support as well; I know other Bitcoin derivatives that are Lightning-compatible (e.g. Groestlcoin) are going to want to use this to support their coins as well.

Currently, the code takes its username/password configuration from the same options as btcd; if they aren't specified, it first searches for a cookie file and then searches for a specified user/password in `bitcoin.conf`. It doesn't support reading `rpcauth` since there's no way to reverse the salted password hash.

Reply:
Sure, but the structure+naming are just inconsistent atm. To an outsider, that doesn't colloquially refer to `bitcoind` as "core" this naming is confusing. Using `bitcoind` is much clearer. We don't need to worry about adding other random alts at this time. The primary goal here is to have a consistent set of arguments and configuration parameters.

Reply:
I've updated the configuration options as we discussed.